src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs

```diff
@@ -452,47 +450,37 @@ private void SessionEndedHandler(object sender, SessionEndEventArgs args)
                     this.logger.LogWarning(this.context.SessionDataCollectionContext, Resources.Resources.NotGeneratingSequenceFile);
                 }
 
-                if (this.collectProcessDumpOnTrigger)
+                if (this.uploadDumpFiles)
                 {
-                    // If there was a test case crash or if we need to collect dump on process exit.
-                    //
-                    // Do not try to collect dump when we already collected one from the hang dump
-                    // we won't dump the killed process again and that would just show a warning on the command line
-                    if ((this.testStartCount > this.testEndCount || this.collectDumpAlways) && !this.dumpWasCollectedByHangDumper)
+                    try
                     {
-                        if (this.uploadDumpFiles)
+                        var dumpFiles = this.processDumpUtility.GetDumpFiles();
+                        foreach (var dumpFile in dumpFiles)
                         {
-                            try
+                            if (!string.IsNullOrEmpty(dumpFile))
                             {
-                                var dumpFiles = this.processDumpUtility.GetDumpFiles();
-                                foreach (var dumpFile in dumpFiles)
+                                try
```

Comment:
nit: It would be nice to have the contents of the `foreach` loop in a separate method, something like `TransferDumpFile`.

Reply:
I'd love to hear your motivation for that. Won't fix right now.